js_modules/dagster-ui/packages/ui-components/src/components/ListItem.tsx

```diff
@@ -0,0 +1,40 @@
+import {ChangeEvent, HTMLProps, ReactNode, forwardRef} from 'react';
+
+import {Checkbox} from './Checkbox';
+import styles from './css/ListItem.module.css';
+
+interface Props {
+  index: number;
+  checked?: boolean;
+  onToggle?: (checked: boolean) => void;
+  href: string;
+  renderLink: (props: HTMLProps<HTMLAnchorElement>) => ReactNode;
```

Comment:
Feels a little bad and I could avoid this by moving the component into `ui-core`. Lmk if you have strong opinions.